pkg/cluster/config_manager.go

```diff
@@ -135,6 +135,10 @@ func (c *ConfigManager) unmarshal(yamlManifest []byte) (*parsed, error) {
 			continue
 		}
 
+		if k.APIVersion == "" {
```

Comment:
I think this is how we should do it for the APIVersion at least. There wouldn't be any way to validate it from the current spec or CRD definition, since it's actually marked as "optional" in the TypeMeta [struct](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#TypeMeta.APIVersion) with "omitempty" for json encoding.

However, in the context of our parsing logic, it's not optional since it's used in the key for when building a map of the parsed objects read from the yaml (which we later use to lookup the objects).